components/social/org.wso2.carbon.social.core/src/main/java/org/wso2/carbon/social/core/ActivityPublisher.java

```diff
@@ -57,9 +57,23 @@ public long publish(String activity) throws SocialActivityException {
 		return publishActivity(jsonObject);
 	}
 
+	public String update(String activity) throws SocialActivityException {
+		JsonObject jsonActivity;
+		try {
+			jsonActivity = (JsonObject) parser.parse(activity);
+		} catch (JsonSyntaxException e) {
+			String message = "Malformed JSON element found: " + e.getMessage();
+			throw new SocialActivityException(message, e);
```

Comment:
no need to add e.getMessage(). it is already there in e. instead put meaningful message.
no need message variable, since it is only used in the exception.